.github/workflows/build-windows.yml

```
@@ -0,0 +1,117 @@
+name: COLMAP (Windows)
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [ assigned, opened, synchronize, reopened ]
+  release:
+    types: [ published, edited ]
+
+jobs:
+  build:
+    name: ${{ matrix.config.os }} ${{ matrix.config.cmakeBuildType }}
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      matrix:
+        config: [
+          {
+            os: windows-2019,
+            cmakeBuildType: Release,
+          },
+          {
+            os: windows-2022,
+            cmakeBuildType: Release,
+          },
+        ]
+
+    env:
+      COMPILER_CACHE_VERSION: 1
+      COMPILER_CACHE_DIR: ${{ github.workspace }}/compiler-cache
+      CCACHE_DIR: ${{ github.workspace }}/compiler-cache/ccache
+      CCACHE_BASEDIR: ${{ github.workspace }}
+      VCPKG_COMMIT_ID: fa6e6a6ec3224f1d3697d544edef6272a59cd834
+      VCPKG_BINARY_SOURCES: "clear;x-gha,readwrite"
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Export GitHub Actions cache env
+        uses: actions/github-script@v6
```

```suggestion
        uses: actions/github-script@v7
```
To suppress the CI warnings